refactor: Added the servicer/router layer separation

mohamad-jaafar-monty opened the pull request:
refactor: Added the servicer/router layer separation

Summary:
- Added `DeleteGitTokenRequest` to serve as the request holder. 

- Added the `delete_by_id_and_user_id` repository method, and added it to the FakeGitLabelStore.

- Added an empty default `__init__` dunder method for repository classes `TortoiseRepoStore`, `TortoiseUserStore`, `TortoiseGitLabelStore`, because when using it with Depends(), FastAPI dependency mechanism will automatically assume its in this form and  cause args and kwargs to show as required query parameters.
  ```
  def __init__(self, *args, **kwargs):
      pass
  ```


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added the ability to delete Git labels by ID for authenticated users.
  * Introduced a new request schema and service to support Git label deletion.

* **Bug Fixes**
  * Updated error response for missing token values to return a 400 Bad Request.

* **Refactor**
  * Streamlined the Git label deletion endpoint to use a dedicated service and request schema.
  * Improved dependency injection handling for repository classes.

* **Tests**
  * Added tests for Git label deletion method covering input validation and deletion outcomes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
refactor: Added the servicer/router layer separation

Summary:
- Added `DeleteGitTokenRequest` to serve as the request holder.

- Added the `delete_by_id_and_user_id` repository method, and added it to the FakeGitLabelStore.

- Added an empty default `__init__` dunder method for repository classes `TortoiseRepoStore`, `TortoiseUserStore`, `TortoiseGitLabelStore`, because when using it with Depends(), FastAPI dependency mechanism will automatically assume its
  ```
  def __init__(self, *args, **kwargs):
      pass
  ```
 Causing args and kwargs to show as required query parameters.

Hayat55 commented on devdox/app/repositories/git_label_repository.py:
```diff
@@ -103,20 +118,40 @@ async def get_by_user_id_and_label(self, offset, limit, user_id, label: str) ->
     async def create_new(self, label_model: GitLabelDBCreateDTO) -> GitLabel:
         return await GitLabel.create(**label_model.model_dump())
+
+    async def delete_by_id_and_user_id(self, label_id:uuid, user_id:str) -> GitLabel | None:
```
I am a bit confused about the expected return type. Should we return retrieved_git_label? If not, would it be better to return a boolean indicating whether the item was found and successfully deleted?

mohamad-jaafar-monty replied:
My first implementation of this method returned an integer that is the total number of rows that were affected by the Delete method, but then after some deliberation, I think a better option is to retrieve the data slated for deletion, and then, I delete them, and I return back what was deleted, for auditing purposes or if you want to do additional operations afterwards.

mohamad-jaafar-monty replied:
resolved in [d95dc02](https://github.com/montymobile1/devdox-ai-portal-api/pull/69/commits/d95dc02aaca1c6abdb93cc9139c58d6b7ad67713)

Commit:
refactor: changed what was requested by the reviewer

Summary:
- Changed the return type and internal functionality for delete git label by id and user id.

- updated the Fake Object to be up to date with the new changes

- Added tests for the repository.
